bin/saya/src/persistent.rs

```diff
@@ -80,10 +87,30 @@ impl Persistent {
 
 impl Start {
     pub async fn run(self) -> Result<()> {
-        let mut snos_file = std::fs::File::open(self.snos_program)?;
+        let mut snos_file = std::fs::File::open(self.snos_program.clone())?;
         let mut snos = Vec::with_capacity(snos_file.metadata()?.len() as usize);
         snos_file.read_to_end(&mut snos)?;
 
+        let saya_path = if let Some(db_dir) = self.db_dir {
+            format!("{}/saya.db", db_dir.display())
+        } else {
+            "saya.db".to_string()
+        };
+
+        let workers_distribution: [usize; 3] =
```

Comment:
```suggestion
        let workers_distribution: [usize; NUMBER_OF_STAGES] =
```
this constant may be changed to `pub` and re-used here.